src/stream.rs

```diff
@@ -1,101 +1,205 @@
 use std::io::{Read, Seek};
-use std::sync::{Arc, Weak};
+use std::marker::Sync;
+use std::sync::Arc;
 use std::{error, fmt};
 
 use crate::decoder;
-use crate::dynamic_mixer::{self, DynamicMixerController};
+use crate::dynamic_mixer::{mixer, Mixer, MixerSource};
 use crate::sink::Sink;
-use crate::source::Source;
 use cpal::traits::{DeviceTrait, HostTrait, StreamTrait};
-use cpal::{Sample, SupportedStreamConfig};
+use cpal::{
+    BufferSize, ChannelCount, FrameCount, Sample, SampleFormat, SampleRate, StreamConfig,
+    SupportedBufferSize,
+};
 
-/// `cpal::Stream` container. Also see the more useful `OutputStreamHandle`.
+const HZ_44100: cpal::SampleRate = cpal::SampleRate(44_100);
```

Comment:
> I'd even make this a global constant that can be used everywhere in rodio since this is de-facto preferred one.

Agreed, but lets make it something like `DEFAULT_SAMPLE_RATE`.

> Some filters use 48kHz which will require unnecessary sample rate conversions.

Thats bad, good catch lets fix that, can you tell me where you saw that? I can not find it myself atm.

> In a more radical change this may be an output stream parameter that can be hinted to the to the filters at creation time so they can adjust their calculations accordingly.

Filters should be able to pick this up already right? Sample rate is part of the Source trait.

Reply:
`SineWave` uses 48kHz [src/source/sine.rs](https://github.com/RustAudio/rodio/blob/489e184a58c3655e7806318d2a8a7248d5055e67/src/source/sine.rs#L17)

Reply:
Sources can tell which rate they are actually using, this is an "output" parameter. Maybe in most cases it is not a problem since filters may use the same rate as their input, but in case of generative ones, they should have some configuration parameter to pick sample rate.

Reply:
> `SineWave` uses 48kHz [src/source/sine.rs](https://github.com/RustAudio/rodio/blob/489e184a58c3655e7806318d2a8a7248d5055e67/src/source/sine.rs#L17)

A yes, thats still in here, I'll put it on the list to deprecate. It has been replaced by signal_generator.rs

Reply:
> but in case of generative ones, they should have some configuration parameter to pick sample rate.

Agreed, but only sine.rs misses that right, and sine is getting ripped out in the same release as this PR. The others do as far as I can see.